Packs/PcapAnalysis/Scripts/PcapFileExtractStreams/PcapFileExtractStreams.py

```diff
@@ -0,0 +1,364 @@
+            port_range = range_str.split('-')
+            if len(port_range) == 1:
+                server_ports.append(((int(port_range[0].strip())), (int(port_range[0].strip()))))
+            elif len(port_range) == 2:
+                server_ports.append(((int(port_range[0].strip())), (int(port_range[1].strip()))))
+            else:
+                raise ValueError(f'Invalid port range: {range_str}')
+
+        rsa_key_file_path = None
+        if rsa_decrypt_key_entry_id:
+            rsa_key_file_path = demisto.getFilePath(rsa_decrypt_key_entry_id).get('path')
+
+        parser = PcapParser(server_ports=server_ports)
+        streams = parser.parse_file(
+            pcap_file_path=demisto.getFilePath(pcap_entry_id).get('path'),
+            wpa_password=wpa_password,
+            rsa_key_file_path=rsa_key_file_path,
+            pcap_filter=pcap_filter
+        )
+        contents = streams.build(bin2txt_mode=bin2txt_mode, filter_keys=filter_keys)
+
+        # Create a summary entry
+        table_headers = ['protocol', 'client_ip', 'client_port', 'server_ip', 'server_port',
+                         'stream_size', 'outgoing_size', 'incoming_size']
+        if filter_keys:
+            table_headers = [k for k in table_headers if k in filter_keys]
+
+        if not table_headers:
+            readable_output = 'No summary'
+        else:
+            readable_output = tableToMarkdown(
+                'PCAP Stream Summary',
+                [{k: stringEscapeMD(str(ent[k]), True, True) for k in ent.keys() & table_headers} for ent in contents],
+                headers=table_headers,
+                headerTransform=lambda s: stringEscapeMD(string_to_table_header(s), True, True))
+        return_results(CommandResults(outputs_prefix='PCAPStream',
+                                      outputs=contents,
+                                      readable_output=readable_output,
+                                      raw_response=contents))
+
+        # Create stream entries
+        if verbose and\
+           contents and\
+           (not filter_keys or ({'stream_text', 'outgoing_text', 'incoming_text'}.intersection(filter_keys))):
+            table_headers = ['protocol', 'client_ip', 'client_port', 'server_ip', 'server_port',
+                             'stream_size', 'outgoing_size', 'incoming_size',
+                             'stream_text', 'outgoing_text', 'incoming_text']
+            if filter_keys:
+                table_headers = [k for k in table_headers if k in filter_keys]
+
+            for ent in contents:
+                readable_output = tableToMarkdown(
+                    'PCAP Stream',
+                    {k: stringEscapeMD(str(ent[k]), True, True) for k in ent.keys() & table_headers},
+                    headers=table_headers,
+                    headerTransform=lambda s: stringEscapeMD(string_to_table_header(s), True, True))
+                return_results(CommandResults(readable_output=readable_output))
+
```

Comment:
I'm not sure if this block of code is reachable... I think the main function will be ended in the return_results of line 336.

Reply:
Yes, that is an issue.

line 336 needs to be changed to:

`        demisto.results(CommandResults(outputs_prefix='PCAPStream',
                                      outputs=contents,
                                      readable_output=readable_output,
                                      raw_response=contents).to_context())`

Reply:
This is ok and line 336 will not prevent the continue of the run. @spearmin10 would you still want us to change it?

Reply:
@JasBeilin Got it. I've checked that `return_results` simply sets resutls into `demisto.results` and doesn't exit. It will pass though the following codes. It doesn't need to be changed.